File: Programmers/완주하지 못한 선수.js
const participant = ["stanko", "mislav", "ana"];
const completion = ["stanko", "ana", "mislav"];

function solution(par, com) {
    let parTemp = '';
    for(let i = 0, count = 0; ; i++) {
        parTemp = par.shift();
        if(parTemp === com[count]) {
            count++;
            if(count === com.length) {
                break;
            }
        } else {
            par.push(parTemp);
        }
    }

    return par;
}

const answer = solution(participant, completion);
console.log(answer);

File: Programmers/완주하지_못한_선수.py
participant = ['leo', 'jin', 'par']
completion = ['leo', 'par']


def solution(participant, completion):
    temp = 0
    hash_table = {}

    for p in participant:
        hash_table[hash(p)] = p
        temp += int(hash(p))

    for c in completion:
        temp -= int(hash(c))

    answer = hash_table[temp]
    print(hash_table)
    return answer


print(solution(participant, completion))
